src/inmanta/loader.py

```diff
@@ -250,14 +252,19 @@ def touch_inits(directory: str) -> None:
                 normdir: str = os.path.normpath(directory)
                 if normdir == package_dir:
                     return
-                pathlib.Path(os.path.join(normdir, init_file)).touch()
+                pathlib.Path(os.path.join(normdir, "__init__.py")).touch()
                 touch_inits(os.path.dirname(normdir))
 
             # ensure correct package structure
             os.makedirs(module_dir, exist_ok=True)
             touch_inits(os.path.dirname(module_dir))
             source_file = os.path.join(module_dir, init_file)
 
+            if os.path.exists(os.path.join(module_dir, "__init__.py")) and source_file[-4:] == ".pyc":
+                # we seem to have create an __init__.py before, we now have a pyc file so we should remove it before
+                # creating the pyc to keep the structure clean
```

Comment:
Nice, this does indeed resolve my concerns in this direction. I see how the finder prefers `.py` over `.pyc`, which resolves the concern in the other direction. But wouldn't it be safer to just do the opposite here as well? Then we can never have both files simultaneously, and no assumptions are required between the finder and the loader.

Reply:
This comment is not wholly accurate: the finder is not used on the agent iirc, it's just native Python. But I still think it might be best to play safe.